runelite-client/src/main/java/net/runelite/client/plugins/slayer/TargetClickboxOverlay.java

```diff
@@ -26,59 +26,78 @@
  */
 package net.runelite.client.plugins.slayer;
 
-import java.awt.BasicStroke;
 import java.awt.Color;
 import java.awt.Dimension;
 import java.awt.Graphics2D;
 import java.awt.Polygon;
 import java.util.List;
 import javax.inject.Inject;
+import net.runelite.api.Client;
 import net.runelite.api.NPC;
+import net.runelite.api.Perspective;
+import net.runelite.api.coords.LocalPoint;
 import net.runelite.client.ui.overlay.Overlay;
 import net.runelite.client.ui.overlay.OverlayLayer;
 import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayUtil;
 
 public class TargetClickboxOverlay extends Overlay
 {
 	private final SlayerConfig config;
 	private final SlayerPlugin plugin;
-
+	private final Client client;
 	@Inject
-	TargetClickboxOverlay(SlayerConfig config, SlayerPlugin plugin)
+	TargetClickboxOverlay(SlayerConfig config, SlayerPlugin plugin, Client client)
 	{
 		this.config = config;
 		this.plugin = plugin;
+		this.client = client;
 		setPosition(OverlayPosition.DYNAMIC);
 		setLayer(OverlayLayer.ABOVE_SCENE);
 	}
 
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		if (!config.highlightTargets())
-		{
-			return null;
-		}
-
 		List<NPC> targets = plugin.getHighlightedTargets();
-		for (NPC target : targets)
+		if (config.highlightStyle() == HighlightStyle.NONE)
 		{
-			renderTargetOverlay(graphics, target, config.getTargetColor());
+			return null;
 		}
-
+		renderTargetList(graphics, targets, config.getTargetColor(), config.highlightStyle());
 		return null;
 	}
-
-	private void renderTargetOverlay(Graphics2D graphics, NPC actor, Color color)
+	private void renderTargetList(Graphics2D graphics, List<NPC> targets, Color highlightColor, HighlightStyle style)
 	{
-		Polygon objectClickbox = actor.getConvexHull();
-		if (objectClickbox != null)
+		if (targets != null)
 		{
-			graphics.setColor(color);
-			graphics.setStroke(new BasicStroke(2));
-			graphics.draw(objectClickbox);
-			graphics.setColor(new Color(color.getRed(), color.getGreen(), color.getBlue(), 20));
-			graphics.fill(objectClickbox);
+			for (NPC npc : targets)
+			{
+				if (npc != null)
+				{
+					switch (style)
+					{
+						case NONE:
```

Comment:
this entire switch is still mostly exactly same as in NPCIndicators. I would make common method in overlayUtil where you pass like Actor and the HighlightStyle, and make both NPC indicators and slayer plugin is same HighlightStyle class